### PR TITLE
Parameterize no log flag

### PR DESCRIPTION
**Pull request by david-dc-chen**
Parameterize no_log attribute so that it can be override at command line as needed for easy testing.
By default it won't log sensitive info.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,14 +116,10 @@ For example: -vvv for more, -vvvv to enable connection debugging.
...
-For debugging purpose, you can change setting to "no_log: false " temporarily to enable Ansible output on your terminal. 
-Remember to set back to true when debugging is done.
+To keep sensitive values such as decrypted password out of your logs, tasks that expose them are marked with the no_log attribute and set to true by default.
+For debugging purpose, you can override "no_log" default at the command line to enable Ansible output on your terminal temporarily. 
 ```
-- name: Dump gateway source database
-  no_log: true # don't log decrypted passwords in responses
-  shell: 
-    cmd: mysqldump {{ database_name }} --routines -u{{ database_user}}  -p{{ database_pass }} > {{remote_db_temp_dir}}/ssg.sql
+ansible-playbook playbooks/gateway-autoprovision-nodes.yml --extra-vars no_log=flase
```
Comment (jagsgill) [inline README.md:122]: typo `flase`

File: roles/gateway_common/tasks/initial-update-root-password.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 - name: Update expired root user password
-  no_log: true # don't log decrypted passwords in responses
+  no_log: "{{ no_log }}" # don't log decrypted passwords in responses
```
Comment (jagsgill) [inline roles/gateway_common/tasks/initial-update-root-password.yml:3]: @david-dc-chen I did this with another variable and Ansible had a stack overflow error. This one works?

Comment (david-dc-chen) [reply roles/gateway_common/tasks/initial-update-root-password.yml:3]: I tried with gateway-autoprovision-nodes.yml  and it works there. I've used {{ no_log_flag }} in the past and it worked too at time.

Comment (jagsgill) [reply roles/gateway_common/tasks/initial-update-root-password.yml:3]: Hmm ok, might still wanna pick a different variable name though. It's a bit confusing as a reader to see `x = x`.

**Pull request merged**